src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Windows.cs

```diff
@@ -342,12 +342,19 @@ internal unsafe int IndexOfCore(string source, string target, int startIndex, in
 
             if ((options & CompareOptions.Ordinal) != 0)
```

Comment:
Does `CompareInfo.Unix.cs` need the same change for parity? 

It seems there is opportunity for refactoring. It does not look right for this platform neutral code to be in platform-specific file.

Reply:
Moved `IndexOfCore` into `CompareInfo.cs` and added `IndexOfPlatform` for `CompareInfo.Unix.cs` and `CompareInfo.Windows.cs`